src/Mapping/Attribute/Document.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\ODM\MongoDB\Mapping\Attribute;
+
+use Attribute;
+
+/**
+ * Identifies a class as a document that can be stored in the database
+ *
+ * @final
+ */
+#[Attribute(Attribute::TARGET_CLASS)]
+class Document extends AbstractDocument implements MappingAttribute
```

Comment:
Let's move this to `AbstractDocument`, `AbstractField`, `AbstractIndex`:
```suggestion
class Document extends AbstractDocument
```
Less things to remember about in case we add another attribute which extends that class.